upb/def.h

```diff
@@ -307,6 +307,12 @@ class upb::FieldDef {
   uint32_t number() const;   /* Returns 0 if uninitialized. */
   bool is_extension() const;
 
+  /* Get the JSON name for this field.  This will copy the JSON name into the
+   * given buffer, which must have size of at least "strlen(name()) + 1".
+   * The string will be NULL-terminated.  Returns false if uninitialized.
+   */
+  bool GetJsonName(char* buf) const;
```

Comment:
C++ api seems like it would be easier to use if it took a string (and called reserve on it if current capacity is not enough)